### ISSUE TITLE
Hide the OTP code with asterisks/dots

### ISSUE THREAD

**Issue by faramonius** [Kunzisoft/KeePassDX#1851]
**Is your feature request related to a problem? Please describe.**

Sometimes it happens that we copy the OTP code while in a crowd or under surveillance cameras. The desire not to show the code to other people is a normal desire from a privacy point of view. Even if the code changes every 30 seconds, I want to be able to copy it without being able to see it.

**Describe the solution you'd like**

I suggest adding the optional ability to mask the OTP code for 2FA identification behind asterisks or dots - just like you do for passwords or secure records. You could optionally add a "view" button to the field for the OTP code in the settings. 

**Describe alternatives you've considered**

No.

**Additional context**

No.

**Comment by landry314** [comment]
But in less than 30 seconds it will be useless information. I can't imagine how someone looking at it could use it fast enough to hurt you.

**Comment by J-Jamet** [comment]
OTP token hidden by asterisks by default:
 - A single tap displays the code and copies it to the clipboard
 - If a code reaches the end of its time, the new code is replaced by asterisks
 - A long press keeps the code hidden and copies it to the clipboard
Is that OK?

**Comment by faramonius** [comment]
@J-Jamet 
I think it's better to do it the other way around: a single tap copies the invisible code hidden by asterisks, and a long tap copies and displays it.

I'd prefer to add another button - one that would simply display the code, without copying it, so I could enter it manually on another device.
If the code is hidden by default, this button will be useful.
If you add this button, you won't need to do a longtap.

Thanks!

**Comment by J-Jamet** [comment]
OK, I'll reverse the behavior of single-click and long-click. The visibility button will of course be added to the entry screen, but I won’t be adding it to the group screen that lists entries with OTP, that’s why I asked what you preferred. The behavior of the “Show OTP Token” setting will be changed to include or exclude asterisks, with visibility maintained in all cases for the timer.

**Comment by faramonius** [comment]
Excuse me, is it possible to make the totp field hidden by dots in an open entry (where logins, passwords, and TOTP are) by default? Right now, by default, I see numbers when I open each entry. I can tap the "hide" button, but when I reopen it, I see the numbers again. There's no option in the program settings to change this behavior. I think it should work the same way as with passwords: the totp numbers should be hidden by dots by default and then shown on demand, by taping a button.

**Comment by J-Jamet** [comment]
Yes it's a bug : https://github.com/Kunzisoft/KeePassDX/issues/2665
